pkg/controllers/secret/secret_controller.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2022 pbrookes@redhat.com.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secret
+
+import (
+	"context"
+	"net/url"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/multiClusterWatch"
+)
+
+// SecretReconciler reconciles a Secret object
+type SecretReconciler struct {
+	client.Client
+	Scheme  *runtime.Scheme
+	MCWatch multiClusterWatch.Interface
+}
+
+type TLSClientConfig struct {
+	Insecure bool   `json:"insecure"`
+	CaData   []byte `json:"caData,omitempty"`
+	CertData []byte `json:"certData,omitempty"`
+	KeyData  []byte `json:"keyData,omitempty"`
+}
+
+type ProviderConfig struct {
+	Command    string   `json:"command,omitempty"`
+	Args       []string `json:"args,omitempty"`
+	APIVersion string   `json:"apiVersion,omitempty"`
+}
+
+type ArgoClusterConfig struct {
+	BearerToken        string          `json:"bearerToken,omitempty"`
+	Username           string          `json:"username,omitempty"`
+	Password           string          `json:"password,omitempty"`
+	TlsClientConfig    TLSClientConfig `json:"tlsClientConfig,omitempty"`
+	ExecProviderConfig ProviderConfig  `json:"execProviderConfig,omitempty"`
+}
+
+const (
+	ARGO_CLUSTER_LABEL       = "argocd.argoproj.io/secret-type"
```

Comment:
Yeah I think, while we can support the format we should prob have our own label, WDYT? it can be a follow on issue though